src/wazuh_modules/agent_upgrade/wm_agent_parsing.c

```diff
@@ -0,0 +1,134 @@
+/*
+ * Wazuh Module for Agent Upgrading
+ * Copyright (C) 2015-2020, Wazuh Inc.
+ * July 3, 2020.
+ *
+ * This program is free software; you can redistribute it
+ * and/or modify it under the terms of the GNU General Public
+ * License (version 2) as published by the FSF - Free Software
+ * Foundation.
+ */
+#include "wm_agent_parsing.h"
+#include "wm_agent_upgrade_tasks.h"
+
+cJSON* wm_agent_parse_command(const char* buffer) {
+    cJSON *json_api = NULL; // Response for API
+    cJSON * root = cJSON_Parse(buffer);
+    if (!root) {
+        mterror(WM_AGENT_UPGRADE_LOGTAG, WM_UPGRADE_JSON_PARSE_ERROR,  buffer);
+        json_api = wm_agent_parse_response_mesage(PARSING_ERROR, upgrade_error_codes[PARSING_ERROR], NULL, NULL, NULL);
+    } else {
+        cJSON *params = cJSON_GetObjectItem(root, "params");
+        const char *command = cJSON_GetObjectItem(root, "command")->valuestring;
+        if (strcmp(command, WM_AGENT_UPGRADE_COMMAND_NAME) == 0) {
+            json_api = wm_agent_process_upgrade_command(params, cJSON_GetObjectItem(root, "agents"));
+        } else if (strcmp(command, WM_AGENT_UPGRADE_RESULT_COMMAND_NAME) == 0) { 
+            json_api = wm_agent_process_upgrade_result_command(cJSON_GetObjectItem(root, "agents"));
+        } else {
+            // TODO invalid command
+            mterror(WM_AGENT_UPGRADE_LOGTAG, WM_UPGRADE_UNDEFINED_ACTION_ERRROR,  command);
+            json_api = wm_agent_parse_response_mesage(TASK_CONFIGURATIONS, upgrade_error_codes[TASK_CONFIGURATIONS], NULL, NULL, NULL);
+        }
+        cJSON_Delete(root);
+    }
+    return json_api;
+}
+
+wm_upgrade_task* wm_agent_parse_upgrade_command(const cJSON* params, char* output) {
+    wm_upgrade_task *task = wm_agent_init_upgrade_task();
+    int param_index = 0;
+    int error_flag = 0;
+    while(!error_flag && (param_index < cJSON_GetArraySize(params))) {
+        cJSON *item = cJSON_GetArrayItem(params, param_index++);
```

Comment:
Add a check to make sure that `item` is not NULL.

Reply:
In this case there is no need to. In the case that there is no items in the array `cJSON_GetArraySize`() would return 0 and it will not enter the while loop